src/librustc_mir/const_eval/eval_queries.rs

```diff
@@ -57,6 +57,12 @@ fn eval_body_using_ecx<'mir, 'tcx>(
     ecx.run()?;
 
     // Intern the result
+    // FIXME: since the DefId of a promoted is the DefId of its owner, this
+    // means that promoteds in statics are actually interned like statics!
+    // However, this is also currently crucial because we promote mutable
+    // non-empty slices in statics to extend their lifetime, and this
+    // ensures that they are put into a mutable allocation.
+    // For other kinds of promoteds in statics (like array initializers), this is rather silly.
```

Comment:
Now you had me worried if I can put mutable references in array initializers, but no:

```rust
#![feature(const_in_array_repeat_expressions)]

static mut FOO: [&mut i32; 3] = [&mut 42; 3];
```

fails to compile